main.tf

```diff
@@ -1,3 +1,14 @@
+locals {
+  all_secrets = merge(
+        var.secret_environment,
+        var.secret_environment,
+      )
+}
+
+locals {
+  has_secrets = length(local.all_secrets) > 0
```

Comment:
You can put multiple vars in one locals block. Here, I'd stick to just one block

Reply:
done.